doc/metrics.md

```diff
@@ -0,0 +1,286 @@
+
+### Filtering metrics
+
+To get all `Timer` objects registered for a `MetricRegistry`, use the
+`getClientMetrics()` method in the `Metrics` class. This takes the
+`MetricRegistry` as an argument and returns a `ClientTimerContainer` object.
+
+The `ClientTimerContainer` object has three fields - `urlTimers`,
+`urlAndMethodTimers`, and `metricIdTimers`. The list of `URLClientTimers` can
+be retrieved from the `ClientTimerContainer` with the `getUrlTimers()` method.
+A list of `URlAndMethodClientTimers` can be retrieved with the
+`getUrlAndMethod()` method. A list of `MetricIdClientTimer`s can be retrieved
+with the `getMetricIdTimers()` method.
+
+To get all `MetricData` objects, representing data for each `Metric`
+registered for a `MetricRegistry`, use the `getClientMetricsData()` methods in
+the `Metric` class. This takes the `MetricRegistry` as an argument and returns
+a `ClientMetricDataContainer` object. This object has a field for each type of
+metric data - `urlData`, `urlAndMethodData`, and `metricIdData`. Each field
+has an associated getter returning a list of the appropriate data object -
+`UrlClientMetricData`, `UrlAndMethodClientMetricData`, and
+`MetricIdClientMetricData`.
+
+#### Filtering by URL
+
+To get URL metrics and metrics data, use the `getClientMetricsByUrl()` and
+`getClientMetricsDataByUrl()` methods in the `Metric` class.
+
+Both of these take as arguments the `MetricRegistry` and a string url. If no
+url is provided, return all url metrics. If no metrics are registered for that
+url, return an empty list.
+
+`getClientMetricsByUrl()` returns a list of `UrlClientTimer` objects.
+`getClientMetricsDataByUrl` returns a list of `UrlClientMetricData` objects.
+
+#### Filtering by URL and method
+
+To get URL and method metrics and metrics data, use the
+`getClientMetricsByUrlAndMethod()` and `getClientMetricsDataByUrlAndMethod()`
+methods in the `Metric` class.
+
+Both of these take as arguments the `MetricRegistry`, a string url, and a
+string HTTP method. If no url or method is provided, will return all
+url-and-method metrics. If no metrics are registered for that url and method,
+returns an empty list.
+
+`getClientMetricsByUrlAndMethod()` returns a list of `UrlAndMethodClientTimer`
+objects. `getClientMetricsDataByUrlAndMethod` returns a list of
+`UrlAndMethodClientMetricData` objects.
+
+#### Filtering by metric-id
+
+To get metric-id metrics and metrics data, use the
+`getClientMetricsByMetricId()` and `getClientMetricsDataByMetricId()` methods
+in the `Metric` class.
+
+Both of these take as arguments the `MetricRegistry` and a metric-id - as an
+array of strings. If no metric-id is provided, will return all metric-id
+metrics. If no metrics are registered for that metric-id, returns an empty
+list.
+
+`getClientMetricsByMetricId()` returns a list of `MetricIdClientTimer`
+objects. `getClientMetricsDataByMetricId()` returns a list of
+`MetricIdClientMetricData` objects.
+
+## Clojure API
```

Comment:
I think I'd probably more the Clojure API section up above the Java one, since more of our devs will be using it.